packages/wallets-helpers/src/factories/walletConnectLegacyConnector.ts

```diff
@@ -0,0 +1,36 @@
+import { WalletConnectLegacyConnector } from 'wagmi/connectors/walletConnectLegacy';
+import { isValidHttpUrl } from '../utils';
+import { walletConnectMobileLinks } from './walletConnectMobileLinks';
+
+export const prepareWalletConnectRPC = (rpc: Record<number, string>) => {
+  const BASE_URL = typeof window === 'undefined' ? '' : window.location.origin;
+  // adds BASE_URL to `rpc` object's string values
+  return Object.entries(rpc).reduce(
+    (result, [key, value]) => ({
+      ...result,
+      [key]: isValidHttpUrl(value) ? value : BASE_URL + value,
+    }),
+    {} as typeof rpc,
+  );
```

Comment:
Instead of "isValidHttpUrl" you can use `new Url(value, BASE_URL).href` which will handle properly both absolute and relative paths.

you can use Object.fromEntries instead, like this

```suggestion
  return Object.fromEntries(
    Object.entries(rpc).map(
      ([key, value]) => [key, new URL(value, BASE_URL).href]
    )
  );
```

And last thing - how this is intended to work on server (for relative URLs)? will not it cause some hydration issues?

Reply:
Agree, but looks like this code will not be needed for WC v2 at all and will be removed soon.